gnome-image-installer/util/gis-unattended-config.c

```diff
@@ -0,0 +1,175 @@
+      break;
+
+    case N_PROPERTIES:
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, property_id, pspec);
+      break;
+    }
+}
+
+static void
+gis_unattended_config_dispose (GObject *object)
+{
+  GisUnattendedConfig *self = GIS_UNATTENDED_CONFIG (object);
+
+  g_clear_pointer (&self->key_file, g_key_file_unref);
+
+  G_OBJECT_CLASS (gis_unattended_config_parent_class)->dispose (object);
+}
+
+static void
+gis_unattended_config_finalize (GObject *object)
+{
+  GisUnattendedConfig *self = GIS_UNATTENDED_CONFIG (object);
+
+  g_clear_pointer (&self->file_path, g_free);
+
+  G_OBJECT_CLASS (gis_unattended_config_parent_class)->finalize (object);
+}
+
+static void
+gis_unattended_config_class_init (GisUnattendedConfigClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->set_property = gis_unattended_config_set_property;
+  // object_class->get_property = gis_unattended_config_get_property;
+  object_class->dispose = gis_unattended_config_dispose;
+  object_class->finalize = gis_unattended_config_finalize;
+
+  props[PROP_FILE_PATH] = g_param_spec_string (
+      "file-path",
+      "file path",
+      "Path to unattended.ini file",
+      NULL,
+      G_PARAM_WRITABLE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+
+  g_object_class_install_properties (object_class, N_PROPERTIES, props);
+}
+
+static gboolean
+gis_unattended_config_initable_init (GInitable    *initable,
+                                     GCancellable *cancellable,
+                                     GError      **error)
+{
+  GisUnattendedConfig *self = GIS_UNATTENDED_CONFIG (initable);
+  g_autoptr(GError) local_error = NULL;
+
+  if (self->file_path == NULL)
+    {
+      g_set_error (error,
+                   G_FILE_ERROR,
+                   G_FILE_ERROR_INVAL,
+                   "programmer error: GisUnattendedConfig:file-path must "
+                   "not be NULL");
+      return FALSE;
+    }
+
+  if (!g_key_file_load_from_file (self->key_file,
+                                  self->file_path,
+                                  G_KEY_FILE_NONE,
+                                  &local_error))
+    {
+      if (!g_error_matches (local_error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
+        {
+          /* Force all other errors into the GIS_UNATTENDED_ERROR domain,
+           * but leave ENOENT untouched so it can be easily caught.
+           */
+          local_error->domain = GIS_UNATTENDED_ERROR;
+          local_error->code = GIS_UNATTENDED_ERROR_READ;
+        }
+
+      g_propagate_error (error, g_steal_pointer (&local_error));
+      return FALSE;
+    }
+
+  return TRUE;
+}
+
+static void
+gis_unattended_config_initable_iface_init (GInitableIface *iface)
+{
+  iface->init = gis_unattended_config_initable_init;
+}
+
+GisUnattendedConfig *
+gis_unattended_config_new (const gchar *file_path,
+                           GError **error)
+{
+  g_return_val_if_fail (file_path != NULL, NULL);
```

Comment:
Nitpick: If you’re going to put preconditions here (which is a good thing), you might as well also put one in for `error`:
```
g_return_val_if_fail (error == NULL || *error == NULL, NULL);
```